[ERROR] AttributeError: 'Pool' object has no attribute 'children'

Odd error to receive, any ideas?

[ERROR] AttributeError: 'Pool' object has no attribute 'children'
Traceback (most recent call last):
  File "/var/task/main.py", line 201, in handler
    unnested_content_data = pool_one.map(unnest_data, content_data)
  File "/var/task/lambda_multiprocessing/main.py", line 315, in map
    return self.starmap(func, zip(iterable), chunksize, callback, error_callback)
  File "/var/task/lambda_multiprocessing/main.py", line 325, in starmap
    results = self.starmap_async(func, iterable, chunksize, callback, error_callback)
  File "/var/task/lambda_multiprocessing/main.py", line 322, in starmap_async
    return [self.apply_async(func, args) for args in iterable]
  File "/var/task/lambda_multiprocessing/main.py", line 322, in <listcomp>
    return [self.apply_async(func, args) for args in iterable]
  File "/var/task/lambda_multiprocessing/main.py", line 305, in apply_async
    for c in self.children:

--- a/lambda_multiprocessing/main.py
+++ b/lambda_multiprocessing/main.py
@@ -245,15 +245,14 @@ class Pool:
 
         self._closed = False
 
-    def __enter__(self):
-        self._closed = False
-
         if self.num_processes > 0:
             self.children = [Child() for _ in range(self.num_processes)]
         else:
             # create one 'child' which will just do work in the main thread
             self.children = [Child(main_proc=True)]
 
+
+    def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):

--- a/lambda_multiprocessing/test_main.py
+++ b/lambda_multiprocessing/test_main.py
@@ -282,6 +282,24 @@ class TestMap(TestCase):
         with Pool() as p:
             p.map(square, range(10**3))
 
+    def test_without_with(self):
+        # check that the standard library Pool
+        # can do .map() without `with`
+        p = multiprocessing.Pool(3)
+        ret = p.map(square, [1,2])
+        self.assertEqual(ret, [1,2*2])
+        p.close()
+        p.join()
+
+        # now check that this library can do it
+        p = Pool(3)
+        ret = p.map(square, [1,2])
+        self.assertEqual(ret, [1,2*2])
+        p.close()
+        p.join()
+
+        
+
 class TestMapAsync(TestCase):
     def test_simple(self):
         args = range(5)
@@ -432,7 +450,7 @@ class TestMoto(TestCase):
         self.assertEqual(ret, data)
 
 class TestSlow(TestCase):
-    @unittest.skip('Very slow')
+    #@unittest.skip('Very slow')
     def test_memory_leak(self):
         for i in range(10**2):
             with Pool() as p:

